File: Data Structures & Algorithms/longest-repeating-substring-with-replacement/submission-0.go
func characterReplacement(s string, k int) int {
    l := 0
    maxFreq := 0

    count := make(map[byte]int)

    res := 0

    for r := 0 ; r < len(s) ; r++ {
        count[s[r]]++

        if count[s[r]] > maxFreq {
            maxFreq = count[s[r]]
        }

        for (r-l+1) - maxFreq > k {
            count[s[l]]--
            l++
        }

        if r-l+1 > res {
            res = r-l+1
        }
        
    }
    return res
}

File: Data Structures & Algorithms/counting-bits/submission-0.rs
impl Solution {
    pub fn count_bits(mut n: i32) -> Vec<i32> {
        let mut res = Vec::new();

        while n >= 0 {
            let mut count = 0;

            for i in 0..32 {
                if n & (1 << i) != 0 {
                    count += 1;
                }
            }

            res.push(count);

            n -= 1;
        }

        res.reverse();
        res
    }
}
